flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mongodb-cdc/src/main/java/org/apache/flink/cdc/connectors/mongodb/table/MongoDBTableSourceFactory.java

```diff
@@ -103,6 +109,18 @@ public DynamicTableSource createDynamicTableSource(Context context) {
                         : ZoneId.of(zoneId);
 
         boolean enableParallelRead = config.get(SCAN_INCREMENTAL_SNAPSHOT_ENABLED);
+
+        // The initial.snapshotting.pipeline config is only used in Debezium mode and
+        // cannot be used in incremental snapshot mode because the semantic is inconsistent.
+        // The reason is that in snapshot phase of incremental snapshot mode, the oplog
+        // will be backfilled after each snapshot to compensate for changes, but the pipeline
+        // operations in initial.snapshotting.pipeline are not applied to the backfill oplog,
+        // which means the semantic of this config is inconsistent.
+        checkArgument(
+                !(enableParallelRead && initialSnapshottingPipeline != null),
```

Comment:
Should we also ensure user aren't configuring other `copyExisting` options in incremental snapshot mode?

Reply:
Yes, I think we should check it to avoid misuse, I will fix it soon.